roles/set_scale/tasks/main.yml

```diff
@@ -0,0 +1,76 @@
+    -o=jsonpath='{range .items[?(@.spec.template.spec.providerSpec.value.instanceType=="{{ machineset_instance_type }}")]}{.spec.replicas}{"\n"}{end}'
+  register: machineset_replicas
+
+- name: Calculate sum of replicas of machinesets with type {{ machineset_instance_type }}
+  set_fact:
+    current_replicas_sum: "{{ machineset_replicas.stdout_lines | map('int') | sum(start=0) }}"
+
+- name: Log sum result
+  debug:
+    msg: "Current sum of replicas is {{ current_replicas_sum }}, wanted {{ scale }}"
+
+- when: current_replicas_sum != scale
+  name: Change all {{ machineset_instance_type }} machinesets replicas to have sum {{ scale }}
+  block:
+  - name: Get the names of existing worker machinesets with type {{ machineset_instance_type }}
+    command:
+      oc get machineset -n openshift-machine-api -o=jsonpath='{range .items[?(@.spec.template.spec.providerSpec.value.instanceType=="{{ machineset_instance_type }}")]}{.metadata.name}{"\n"}{end}'
+    register: oc_get_machinesets
+    failed_when: not oc_get_machinesets.stdout
+  - name: Set the replicas of all {{ machineset_instance_type }} machinesets to 0 except the first
+    command: |
+      oc patch machineset -n openshift-machine-api {{ item }} --patch '{"spec": {"replicas": 0 }}' --type merge
+    loop: "{{ oc_get_machinesets.stdout_lines[1:] }}"
+  - name: Get first machineset
+    set_fact:
+      first_machineset: "{{ oc_get_machinesets.stdout_lines[0] }}"
+  - name: Set the replicas of the first {{ machineset_instance_type }} machineset ({{ first_machineset }}) to {{ scale }}
+    command: >
+      oc patch machineset -n openshift-machine-api {{ first_machineset }}
+      --patch '{"spec": {"replicas": {{ scale }} }}' --type merge
+
+- block:
+  - name: Wait for all machinesets with type {{ machineset_instance_type }} to be ready
+    # jq purpose, respectively:
+    # 1. Default .status.availableReplicas to 0 if it's not set, to simplify further queries
+    # 2. Select all machinesets with status fields that don't match current .spec.replicas
+    # 3. Perform some extra formatting for nicer logging
+    shell: >
+      oc get machinesets -n openshift-machine-api \
+        -o=jsonpath='{.items[?(@.spec.template.spec.providerSpec.value.instanceType=="{{ machineset_instance_type }}")]}' \
+      | jq '. as $root | .status | if has("availableReplicas") then $root else $root * {"status": {"availableReplicas": 0 }} end' \
+      | jq 'select(.status.availableReplicas != .spec.replicas or .status.replicas != .spec.replicas)' \
+      | jq '{
+         "name": .metadata.name,
+          "expected_replicas": .spec.replicas,
+          "current_replicas": .status.replicas,
+          "current_available_replicas": .status.availableReplicas
+      }'
+    register: non_ready_replicas
+    until: not non_ready_replicas.stdout_lines
+    retries: 120 # TODO: Does this correlate with {{ scale }}? Probably not, otherwise might need to factor it in
```

Comment:
no, not directly, the machine go live in parallel!

Reply:
That's what I thought, will remove the TODO because your confirmation made me more confident